core/src/main/java/com/yahoo/imapnio/async/internal/ImapAsyncSessionImpl.java

```diff
@@ -279,14 +280,37 @@ public void setDebugMode(@Nonnull final DebugMode newOption) {
 
     @Override
     public ImapFuture<ImapAsyncResponse> execute(@Nonnull final ImapRequest command) throws ImapAsyncClientException {
+        return execute(command, new Consumer<ImapAsyncResponse>() {
+            @Override
+            public void accept(final ImapAsyncResponse imapAsyncResponse) {
+
+            }
+        }, new Consumer<Exception>() {
+            @Override
+            public void accept(final Exception e) {
+
+            }
+        }, new Runnable() {
+            @Override
+            public void run() {
+
+            }
+        });
+    }
+
+    @Override
+    public <T> ImapFuture<ImapAsyncResponse> execute(final ImapRequest command, final Consumer<ImapAsyncResponse> doneCallback,
```

Comment:
Please check out ImapAsyncSessionImplTest.java and add Unit Tests with use of your new execute(...) function.